runtime/compiler/trj9/z/env/J9CPU.cpp

```diff
@@ -209,11 +204,14 @@ CPU::TO_PORTLIB_get390zLinuxMachineType()
    TR_S390MachineType ret_machine = TR_Z10;  /* return value, z/TPF default */
    const int pi1modArraySize = 5;
    char processorName[pi1modArraySize] = {0};
-   struct pi1dt *pid;
-   /* machine hardware name */
-   pid = (struct pi1dt *)cinfc_fast(CINFC_CMMPID);
-   sprintf(processorName, "%4.X", pid->pi1pids.pi1mslr.pi1mod);
-   int machine = atoi(processorName);
+
+   struct utsname info;
+   if (::uname(&info))
+      {
+      return ret_machine;
+      }
+
+   uint32_t machine = atoi(info.machine);
```

Comment:
Just to confirm, does `utsname.machine` return the 4-digit machine code?  When I tried it on RedHat/SLES on Z, I got `s390x` instead.

Reply:
@joransiu 
I created a test program on our platform and here is the output:
CSMP0097I 11.00.55 CPU-B SS-BSS  SSU-HPN  IS-01   
FILE0001I 11.00.55 START OF DISPLAY FROM myprogram
uname machine field response was: 3906-7D1        
END OF DISPLAY+ 
Here is the program:
   struct utsname info;
   if (uname(&info))
      {
      printf("uname call failed!");
      return -1;
      }

   printf("uname machine field response was: %s", info.machine);

Reply:
Hmm after looking at that code again I realized there are two lines from the old way we were doing it that aren't necessary anymore:
   const int pi1modArraySize = 5;
   char processorName[pi1modArraySize] = {0};

I'll remove those and re-push.

Reply:
Thanks @jjohnst-us for confirming!